libraries/Microsoft.Bot.Connector/ConnectorClientEx.cs

```diff
@@ -27,54 +31,50 @@ public ConnectorClient(Uri baseUri, string microsoftAppId = null, string microso
         {
         }
 
-        private HttpClient _originalHttpClient;
-        protected static Lazy<HttpClient> g_httpClient = new Lazy<HttpClient>(() =>
```

Comment:
Maybe I'm not following what's going on here (only on first cup of ☕️), but it looks like we lost the shared `HttpClient` here without the ability to actually specify one upstream? Right now we're still spinning up new `ConnectorClient` instances every time `ProcessActivity` or `ContinueConversation` are called, so we're going to be soft-leaking TCP connections again.

Reply:
yeah, the code I removed was my hack to try to get the serviceClient to use a single http client.  It turns out that hack didn't actually work correctly and was brittle. 

The guidance from the ServiceClient team is that you should use ServiceClient classes as singletons(!) which means all of our sample code is wrong, but that's the change I made in the V3 connector and then changed the Autofac policies around that to only use a single connector and we pumped our perf up to 150 rps from 4. 

So not having the http client goo in here is actually the right thing to do because my hack was not.

---
In reply to: [198888939](https://github.com/Microsoft/botbuilder-dotnet/pull/702#discussion_r198888939) [](ancestors = 198888939)

Reply:
Yes, `ServiceClient` should essentially be treated just like `HttpClient`. We really just need to change our upstream code that currently news one up per-service URL _and_ credentials to use a factory. That would be the "right" fix IMHO and allow us to avoid having to do hacks of passing through a singleton `HttpClient` ourselves instead.